drivers/usb/udc/udc_smartbond.c

```diff
@@ -0,0 +1,1764 @@
+	}
+}
+
+static void handle_bus_reset(struct usb_dc_state *dev_state)
+{
+	uint32_t alt_ev;
+
+	USB->USB_NFSR_REG = 0;
+	USB->USB_FAR_REG = 0x80;
+	USB->USB_ALTMSK_REG = 0;
+	USB->USB_NFSR_REG = NFSR_NODE_RESET;
+	USB->USB_TXMSK_REG = 0;
+	USB->USB_RXMSK_REG = 0;
+	set_nfsr(dev_state, NFSR_NODE_RESET);
+
+	for (int i = 0; i < EP_MAX; ++i) {
+		dev_state->ep_state[1][i].buf = NULL;
+		dev_state->ep_state[1][i].config.stat.busy = 0;
+	}
+
+	LOG_INF("send USB_DC_RESET");
+	udc_submit_event(dev_state->dev, UDC_EVT_RESET, 0);
+	USB->USB_DMA_CTRL_REG = 0;
+
+	USB->USB_MAMSK_REG = USB_USB_MAMSK_REG_USB_M_INTR_Msk | USB_USB_MAMSK_REG_USB_M_FRAME_Msk |
+			     USB_USB_MAMSK_REG_USB_M_WARN_Msk | USB_USB_MAMSK_REG_USB_M_ALT_Msk |
+			     USB_USB_MAMSK_REG_USB_M_EP0_RX_Msk |
+			     USB_USB_MAMSK_REG_USB_M_EP0_TX_Msk;
+	USB->USB_ALTMSK_REG = USB_USB_ALTMSK_REG_USB_M_RESUME_Msk;
+	alt_ev = USB->USB_ALTEV_REG;
+	check_reset_end(dev_state, alt_ev);
+	empty_ep0_queues(dev_state->dev);
+}
+
+static void usb_clock_on(struct usb_dc_state *dev_state)
+{
+	if (atomic_cas(&dev_state->clk_requested, 0, 1)) {
+		clock_control_on(DEVICE_DT_GET(DT_NODELABEL(osc)),
+				 (clock_control_subsys_rate_t)SMARTBOND_CLK_USB);
+	}
+}
+
+static void usb_clock_off(struct usb_dc_state *dev_state)
+{
+	if (atomic_cas(&dev_state->clk_requested, 1, 0)) {
+		clock_control_off(DEVICE_DT_GET(DT_NODELABEL(osc)),
+				  (clock_control_subsys_rate_t)SMARTBOND_CLK_USB);
+	}
+}
+
+static void handle_alt_ev(struct usb_dc_state *dev_state)
+{
+	struct smartbond_ep_state *ep_state;
+	uint32_t alt_ev = USB->USB_ALTEV_REG;
+
+	if (USB->USB_NFSR_REG == NFSR_NODE_SUSPEND) {
+		usb_clock_on(dev_state);
+	}
+	alt_ev = check_reset_end(dev_state, alt_ev);
+	if (GET_BIT(alt_ev, USB_USB_ALTEV_REG_USB_RESET) && dev_state->nfsr != NFSR_NODE_RESET) {
+		handle_bus_reset(dev_state);
+	} else if (GET_BIT(alt_ev, USB_USB_ALTEV_REG_USB_RESUME)) {
+		if (USB->USB_NFSR_REG == NFSR_NODE_SUSPEND) {
+			set_nfsr(dev_state, NFSR_NODE_OPERATIONAL);
+			if (dev_state->ep_state[0][0].buf != NULL) {
+				USB->USB_MAMSK_REG |= USB_USB_MAMSK_REG_USB_M_EP0_RX_Msk;
+			}
+			USB->USB_ALTMSK_REG = USB_USB_ALTMSK_REG_USB_M_RESET_Msk |
+					      USB_USB_ALTMSK_REG_USB_M_SD3_Msk;
+			/* Re-enable reception of endpoint with pending transfer */
+			for (int ep_idx = 1; ep_idx < EP_MAX; ++ep_idx) {
+				ep_state = usb_dc_get_ep_out_state(dev_state, ep_idx);
+				if (!ep_state->config.stat.halted) {
+					start_rx_packet(dev_state, ep_state);
+				}
+			}
+			udc_submit_event(dev_state->dev, UDC_EVT_RESUME, 0);
+		}
+	} else if (GET_BIT(alt_ev, USB_USB_ALTEV_REG_USB_SD3)) {
+		set_nfsr(dev_state, NFSR_NODE_SUSPEND);
+		USB->USB_ALTMSK_REG =
+			USB_USB_ALTMSK_REG_USB_M_RESET_Msk | USB_USB_ALTMSK_REG_USB_M_RESUME_Msk;
+		usb_clock_off(dev_state);
+		udc_submit_event(dev_state->dev, UDC_EVT_SUSPEND, 0);
+	}
+}
+
+static void udc_smartbond_isr(const struct device *dev)
+{
+	struct usb_dc_state *dev_state = dev->data;
```

Comment:
Please name it `struct udc_smartbond_data;` and use `struct udc_smartbond_data *data = udc_get_private(dev);` or struct udc_smartbond_data *priv = udc_get_private(dev);` consistently.

Reply:
dev_state change to data in all places